pkgs/development/compilers/llvm/5/default.nix

```diff
@@ -23,6 +23,7 @@ let
       ln -s "${cc}/lib/clang/${release_version}/include" "$rsrc"
       ln -s "${targetLlvmLibraries.compiler-rt.out}/lib" "$rsrc/lib"
       echo "-resource-dir=$rsrc" >> $out/nix-support/cc-cflags
+      echo "--sysroot=${stdenv.cc.cc}" >> $out/nix-support/cc-cflags
```

Comment:
Maybe this can work. But i think it should be cc not stdenv.cc.cc so that it comes from clang.

Reply:
Long-term I'd rather patch out the compiled-in search paths altogether and figure out a better way to switch between C++ stdlibs. As stands, though, this patch is designed to preserve the existing behaviour of finding gcc via the sysroot and using that to pick up libstdc++. Setting the sysroot to (clang) `cc`, or indeed any valid path, will fix the purity issue, but breaks any builds that expect `clang++` to work out of the box — all C++ programs will need to explicitly include their stdlib as a dependency. Currently `libstdc++` is found by `--gcc-toolchain`, but setting `--sysroot` overrides that, so we just put it back by explicitly setting the sysroot to gcc.

Reply:
Hmm... The wrapper should have handling for this through ```default_cxx_stdlib_compile```. Is that maybe not working as expected:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/cc-wrapper/default.nix#L48

My concern with using ```stdenv``` here is it pull in the parent's stdenv which could mess up things like cross compilation. But I definitely want to find a way to get rid of the ```--gcc-toolchain``` thing in the future, so perhaps we can handle them together.